src/inlay-hints.ts

```diff
@@ -113,70 +116,47 @@ class InlayHintsFeature implements vscodelc.StaticFeature {
     }
   }
 
-  onDidChangeVisibleTextEditors() {
+  onDidCloseTextDocument(document: vscode.TextDocument) {
     if (!this.enabled)
       return;
+    
+    // Drop state for any file that is now closed.
+    const uri = document.uri.toString();
+    const file = this.sourceFiles.get(uri);
+    if (file) {
+      file.inlaysRequest?.cancel()
+      this.sourceFiles.delete(uri);
+    }        
+  }
 
-    const newSourceFiles = new Map<string, FileEntry>();
+  onDidChangeVisibleTextEditors() {
+    if (!this.enabled)
+      return;
 
-    // Rerender all, even up-to-date editors for simplicity
+    // When an editor is made visible we have no inlay hints.
+    // Obtain and render them, either from the cache or by issuing a request.
     this.context.visibleClangdEditors.forEach(async editor => {
```

Comment:
Yes, this potentially saves client-side work.
This doesn't cause any flicker or any extra server requests (since the visible editors have cached hints), so I'm not sure it's worth it.